### PR TITLE
feat:[CDS-79028]: docs for harness approval notification improvement phase 1

### PR DESCRIPTION
**Pull request by naman-goenka**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the CODEOWNERS. 

## What Type of PR is This?

- [ ] Issue
- [x] Feature
- [x] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by brian-f-harness — DISMISSED** [review]

File: docs/platform/approvals/adding-harness-approval-stages.md [hunk 1]
```diff
@@ -182,6 +182,36 @@ You can select one of the following types of expression for user groups:
...
+:::
+
+Approval notifications are sent to each of the configured **User Groups** in the **Approvers** section. User group's notification settings are used while sending notifications. For more information, go to [Edit notification preferences](/docs/platform/role-based-access-control/add-user-groups#edit-notification-preferences).
+
+:::important
+THe PagerDuty notification channel is not currently supported for approval notifications. This means that even if the PagerDuty notification channel is configured in a user group notification setting, approval notifications will not be sent to that channel.
+
+These notifications help approvers to make their decision. There are two kinds of approval notifications:
+
+- `Approval Required`: Notification is sent when approval step starts containing the pipeline name, execution URL, user who triggered the pipeline, stage name, approval message, organization and project details, pipeline start time and the time left in approval expiry.
```
Comment (brian-f-harness) [inline docs/platform/approvals/adding-harness-approval-stages.md:200]:
```suggestion
- `Approval Required`: A notification is sent when the approval step starts containing the pipeline name, execution URL, user who triggered the pipeline, stage name, approval message, organization and project details, pipeline start time, and the time left in approval expiry.
```

File: docs/platform/approvals/adding-harness-approval-stages.md [hunk 2]
```diff
@@ -182,6 +182,36 @@ You can select one of the following types of expression for user groups:
...
+- `Approved or Rejected`: Notification is sent when approval step is approved or rejected. The notification contains the pipeline name, execution URL, step status, stage name, user who triggered the pipeline, organization and project details, pipeline start time, approval activity details containing approver name, action, and the time of the approval or rejection.
```
Comment (brian-f-harness) [inline docs/platform/approvals/adding-harness-approval-stages.md:201]:
```suggestion
- `Approved or Rejected`: A notification is sent when the approval step is approved or rejected. The notification contains the pipeline name, execution URL, step status, stage name, the user who triggered the pipeline, organization, and project details, pipeline start time, approval activity details containing approver name, action, and the time of the approval or rejection.  
```

File: docs/platform/approvals/adding-harness-approval-stages.md [hunk 3]
```diff
@@ -182,6 +182,36 @@ You can select one of the following types of expression for user groups:
...
+
+These notifications help approvers to make their decision. There are two kinds of approval notifications:
+
+- `Approval Required`: Notification is sent when approval step starts containing the pipeline name, execution URL, user who triggered the pipeline, stage name, approval message, organization and project details, pipeline start time and the time left in approval expiry.
+- `Approved or Rejected`: Notification is sent when approval step is approved or rejected. The notification contains the pipeline name, execution URL, step status, stage name, user who triggered the pipeline, organization and project details, pipeline start time, approval activity details containing approver name, action, and the time of the approval or rejection.  
+
+Use the **Include stage execution details in approval** option to include stage execution details in approval notifications. A summary of completed, running and upcoming stages is shown. Execution metadata like service, environment and infrastructure details are also present for CD stages in approval notifications.
```
Comment (brian-f-harness) [inline docs/platform/approvals/adding-harness-approval-stages.md:203]:
```suggestion
Use the **Include stage execution details in approval** option to include stage execution details in approval notifications. A summary of completed, running, and upcoming stages is shown. Execution metadata like service, environment, and infrastructure details are also present for CD stages in approval notifications.
```

**Review by brian-f-harness — DISMISSED** [review]

**Review by deepakputhraya — APPROVED** [review]

**Comment by naman-goenka** [comment]
image
image

**Review by shylajasun — APPROVED** [review]

**Pull request merged**